src/commands/utils/draft.ts

```diff
@@ -0,0 +1,75 @@
+import { Errorable, combine, failed } from "./errorable";
+import { DeploymentSpecType } from "../../webview-contract/webviewDefinitions/draft/types";
+import path from "path";
+import { ShellOptions, exec } from "./shell";
+import { parseJson } from "./json";
+import { dirSync } from "tmp";
+
+type DryRunOutput = {
+    variables: { [variableName: string]: string };
+    filesToWrite: string[];
+};
+
+export type DeploymentFiles = {
+    [T in DeploymentSpecType]: string[];
+};
+
+export async function getDeploymentFilesToWrite(draftBinaryPath: string): Promise<Errorable<DeploymentFiles>> {
+    const destDir = dirSync();
+    const deploymentTypes: DeploymentSpecType[] = ["helm", "kustomize", "manifests"];
+    const filesResults = await Promise.all(
+        deploymentTypes.map((deploymentType) => getDeploymentFiles(draftBinaryPath, deploymentType, destDir.name)),
+    );
+    const filesResult = combine(filesResults);
+    if (failed(filesResult)) {
+        return filesResult;
+    }
+
+    const [helmFiles, kustomizeFiles, manifestsFiles] = filesResult.result;
+    const result: DeploymentFiles = {
+        helm: helmFiles,
+        kustomize: kustomizeFiles,
+        manifests: manifestsFiles,
+    };
+
+    return { succeeded: true, result };
+}
+
+async function getDeploymentFiles(
+    draftBinaryPath: string,
+    deploymentType: DeploymentSpecType,
+    destDir: string,
+): Promise<Errorable<string[]>> {
+    const variables = {
+        APPNAME: "testapp",
+        IMAGENAME: "testimage",
+        IMAGETAG: "latest",
+        NAMESPACE: "testns",
+        PORT: 80,
+        SERVICEPORT: 80,
+    };
+
+    const variableArgs = Object.entries(variables)
+        .map(([key, value]) => `--variable ${key}=${value}`)
+        .join(" ");
+
+    const language = "java"; // So it doesn't attempt to autodetect the language
```

Comment:
This value doesn't actually matter. It's only there to prevent Draft from trying to auto-detect the language (taken directly from the [devx extension](https://github.com/Azure/aks-devx-tools/blob/d17bde8431b604e6355afd55034548c5c3920630/src/commands/runDraftTool/runDraftDeployment.ts#L641))